component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/reporter/log/LogCounterMetric.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.am.analytics.publisher.reporter.log;
+
+import org.apache.log4j.Logger;
+import org.wso2.am.analytics.publisher.exception.MetricReportingException;
+import org.wso2.am.analytics.publisher.reporter.CounterMetric;
+import org.wso2.am.analytics.publisher.reporter.MetricSchema;
+
+import java.util.Map;
+
+/**
+ * Log counter metric class. Intended for testing only.
+ */
+public class LogCounterMetric implements CounterMetric {
+    private static final Logger log = Logger.getLogger(LogCounterMetric.class);
+    private final String name;
+
+    protected LogCounterMetric(String name) {
+        this.name = name;
+    }
+
+    @Override public int incrementCount(Map<String, String> properties) throws MetricReportingException {
```

Comment:
no need a new line after the annotation?

Reply:
Checkstyles didnt catch. But anyway fixed with ae699e7e3b6108f6396fb3b470d5dda7ff226233